src/Microsoft.TestPlatform.Build/Microsoft.TestPlatform.targets

```
@@ -24,12 +28,61 @@ Copyright (c) .NET Foundation. All rights reserved.
     Main entry point for running tests through vstest.console.exe
     ============================================================
   -->
-  <Target Name="VSTest" DependsOnTargets="ShowInfoMessageIfProjectHasNoIsTestProjectProperty">
+  <Target
+    Name="VSTest"
+    DependsOnTargets="ShowInfoMessageIfProjectHasNoIsTestProjectProperty"
+  >
+      <!-- Unloggable but colorized output (cf. https://github.com/microsoft/vstest/issues/680) -->
+      <CallTarget Targets="_VSTestConsole" Condition="$(VSTestUseConsole)" />
+      <!-- Proper MSBuild integration, but no custom colorization -->
+      <CallTarget Targets="_VSTestMSBuild" Condition="!$(VSTestUseConsole)" />
+  </Target>
+
+  <!-- Used when called from dotnet msbuild command line: uses MSBuild logs but does not allow for console colorization -->
+  <Target
+    Name="_VSTestMSBuild"
+    DependsOnTargets="$(_VSTestMSBuildDependsOn)"
+    Condition="'$(IsTestProject)' == 'true'"
+  >
+    <Microsoft.TestPlatform.Build.Tasks.VSTestTask
```

```suggestion
    <VSTestTask
```

This is more than enough!